docs/hugo/content/design/secrets.md

```diff
@@ -0,0 +1,669 @@
+* Cast the provided `obj` to the expected type. This should be guaranteed safe because the method won't have been called unless its implemented.
+* Check the `forOperator` section of the spec to determine if any keys should be written. This includes determining the destination secret name and any
+  annotations or other properties that should be written or updated on that secret. Also includes ensuring that the secret in question (if it exists)
+  is owned by the resource in question.
+* Make the required `GetKeys` or `ListKeys` call to Azure.
+* Create or update the secret.
+
+#### Lifecycle
+These are secrets created by the operator (usually after calling some `ListKeys` type API). Since the these secrets are by definition specific to the resource 
+that created them, their ownership in Kubernetes will be set to the resource that created them. When the owning resource is deleted, the created secret will 
+also be deleted.
+
+#### Open questions
+
+**How does key rollover work for these types of secrets?**
+
+We don't support this at all initially. This is a somewhat advanced scenario that doesn't fit well into the Kubernetes resource model anyway
+because rollover is more of an action and less of an actual resource. The user can roll their secrets using the `az cli` (or other tooling) and 
+then either wait for a reconcile to occur naturally or force one to refresh the secrets locally in the cluster.
+
+When we do decide to support this, we should be able to do so as a `Job`-esque resource that runs to completion (and somehow triggers a re-reconcile
+on the parent resource type).
+
+Conclusion: TBD
+
+**How do deal with soft-delete and purge protection?**
+
+If we support KeyVault, we have to deal with the soft-delete + purge awkwardness that deleting and recreating brings. Some of this we dodge by giving control 
+to the user and expecting them to provide us with the name of a secret that's going to work. We may also need a flag per-secret (or global at the operator 
+level?) for if we should purge the secret when we delete. Something like `purgeOnDelete`.
+
+The main downside of a global flag controlling the behavior of purge or delete is that it doesn't give fine-grain control, yet we are proposing to give 
+fine-grained control over what vault to put secrets in. This seems like a discrepancy. This seems to suggest we just put this option on every KeyVault reference:
+
+```yaml
+spec:
+  # Other spec fields elided...
+  forOperator:
+    secrets:
+      # Save the read-write keys (and endpoints) into a kubernetes secret called "my-secret" and a KeyVault secret called "my-secret".
+      keyDestination:
+        keyVault:
+          reference:
+            armId: /subscriptions/.../resourceGroups/.../providers/Microsoft.KeyVault/vaults/asokeyvault
+          name: my-secret
+          purge: false  # Optional: defaults to false
+          delete: false # Optional: defaults to true
+```
+
+Conclusion: TBD
```

Comment:
The user has deleted the resource the secret corresponds to though - there's no resource left for the secret to correspond to.

Imagine the scenario where the user has a storage account and puts theirs keys into KeyVault. Then they delete their storage account. Does it make sense to leave those keys in KeyVault (and not purge them?). That just means that if you wanted to recreate the storage account you'd need to use a different secret name as the old one is taken for _90 days_ until it is purged.

The keys there have no value though because even if you recreate the exact same named storage account, it'll have different keys.

Still, the proposal here is to by default leave them there (and thus make customers use a different name when recreating), but they can optionally purge them, which is useful especially for cases like testing where you'll probably be creating and deleting similarly named resources and secrets regularly.